src/core/eme/create_session.ts

```diff
@@ -0,0 +1,241 @@
+ * @param initDataType
+ * @param mediaKeysInfos
+ * @returns {Observable}
+ */
+function loadPersistentSession(
+  storedSessionId: string,
+  initData: Uint8Array,
+  initDataType: string,
+  session: MediaKeySession|IMediaKeySession
+) : Observable<IGetSessionEvent> {
+  log.debug("eme: load persisted session", storedSessionId);
+
+  return castToObservable(session.load(storedSessionId))
+    .catch((error) => {
+      log.warn("eme: no data stored for the loaded session.",
+        storedSessionId);
+
+      const loadedSession = $loadedSessions.get(initData, initDataType);
+      if (loadedSession != null) {
+        $loadedSessions.closeSession(loadedSession);
+      }
+      $storedSessions.delete(initData);
+
+      throw error;
+    })
+    .map((success) => {
+      if (success) {
+        $loadedSessions.add(initData, initDataType, session);
+        return {
+          type: "loaded-persistent-session" as "loaded-persistent-session",
+          value: { session },
+        };
+      } else {
+        return {
+          type: "created-session" as "created-session",
+          value: { session },
+        };
+      }
+    }).do(() => $storedSessions.add(initData, session));
+}
+
+/**
+ * Handle cases where session is persistent.
+ *
+ * If a persisted session exists in the store associated to this initData,
+ * we reuse it without a new license request through the `load` method.
+ * If we load a unusable session, create a new session from scratch.
+ *
+ * @param {Uint8Array} initData
+ * @param {string} initDataType
+ * @param {MediaKeySession|Object} session
+ * @param {MediaKeys|Object} mediaKeys
+ */
+function handlePersistentSession(
+  initData: Uint8Array,
+  initDataType: string,
+  session: IMediaKeySession|MediaKeySession,
+  mediaKeys: MediaKeys|IMockMediaKeys
+): Observable<IGetSessionEvent> {
+
+    let persistentSession: MediaKeySession|IMediaKeySession = session;
+    const storedEntry = $storedSessions.get(initData);
+
+    if (storedEntry) {
+      return loadPersistentSession(
+        storedEntry.sessionId, initData, initDataType, session
+      ).map((evt) => {
+        if (
+          evt.type === "loaded-persistent-session" &&
+          !isLoadedSessionUsable(evt.value.session)
+        ) {
+          // Recreate a new session from scratch.
+          $loadedSessions.closeSession(evt.value.session);
+          $storedSessions.delete(new Uint8Array(initData));
+
+          persistentSession =
+            (mediaKeys as any).createSession("persistent-license"); // TS Bug
+
+          $loadedSessions.add(initData, initDataType, session);
+          $storedSessions.add(initData, session);
+
+          return {
+            type: "created-session" as "created-session",
+            value: { session: persistentSession },
+          };
+        }
+        return evt;
+      });
+    }
+
+    $storedSessions.add(initData, session);
+    return Observable.of({
+      type: "created-session" as "created-session",
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=187&issues=AWMmtxj7fKfZeQj7PGdT&open=AWMmtxj7fKfZeQj7PGdT)

[ISSUE_KEY]: AWMmtxj7fKfZeQj7PGdT